### PR TITLE
Accept a tensor set down in the dedicated '.tensor' field.

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe or @geirst  PR

### CODE REVIEW

**Review by geirst — COMMENTED** [review]
Overall looks good, but I would like a unit test for this

**Comment by baldersheim** [comment]
Unit test added.

File: searchlib/src/vespa/searchlib/features/dotproductfeature.cpp [hunk 1]
```diff
@@ -621,9 +633,26 @@ DotProductBlueprint::prepareSharedState(const IQueryEnvironment & env, IObjectSt
...
+        if (attribute->getCollectionType() == attribute::CollectionType::ARRAY) {
+            Property tensorBlob = env.getProperties().lookup(getBaseName(), _queryVector, "tensor");
+            if (attribute->isFloatingPointType() && tensorBlob.found() && !tensorBlob.get().empty()) {
+                const Property::Value & blob = tensorBlob.get();
+                vespalib::nbostream stream(blob.data(), blob.size());
+                if (attribute->getBasicType() == BasicType::FLOAT) {
+                    arguments = std::make_unique<ArrayParam<float>>(stream);
+                } else {
+                    arguments = std::make_unique<ArrayParam<double>>(stream);
+                }
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/features/dotproductfeature.cpp]: here we have successfully extracted the tensor value, but we nevertheless also look up the non-tensor parameter property further down without using it.

**Review by havardpe — COMMENTED** [review]
looks ok, but the setup code could probably be simplified

File: searchlib/src/vespa/searchlib/features/dotproductfeature.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/features/dotproductfeature.cpp]: Good catch, that was not intentional. Fixed now.

**Pull request merged**